pylink/library.py

```diff
@@ -82,11 +83,29 @@ class Library(object):
         'JLINK_SetFlashProgProgressCallback'
     ]
 
-    JLINK_SDK_NAME = 'libjlinkarm'
+    # Linux/MacOS: The JLink library name without any prefix like lib,
+    # suffix like .so, .dylib or version number.
+    JLINK_SDK_NAME = 'jlinkarm'
 
+    # Linux/MacOS: The library file name will start with 'libjlinkarm'
+    # Used by Library.find_library_{linux,darwin}()
+    JLINK_SDK_STARTS_WITH = 'libjlinkarm'
```

Comment:
I have downloaded this version of lib today and my zephyr-os stop working and after some cheking it seams that you need to fix  jlink.py:150  

```
        if not hasattr(self, '_jlink_version'):
            plat = sys.platform
            if plat.startswith('win32'):
                libname = Library.get_appropriate_windows_sdk_name() + '.dll'
            elif plat.startswith('linux'):
                libname = Library.JLINK_SDK_STARTS_WITH + '.so'
            elif plat.startswith('darwin'):
                libname = Library.JLINK_SDK_STARTS_WITH + '.dylib'
            else:
                self.logger.warning(f'unknown platform {plat}; assuming UNIX')
                libname = Library.JLINK_SDK_STARTS_WITH + '.so'
```

Reply:
> I have downloaded this version of lib today and my zephyr-os stop working and after some cheking it seams that you need to fix jlink.py:150
> 
> ```
>         if not hasattr(self, '_jlink_version'):
>             plat = sys.platform
>             if plat.startswith('win32'):
>                 libname = Library.get_appropriate_windows_sdk_name() + '.dll'
>             elif plat.startswith('linux'):
>                 libname = Library.JLINK_SDK_STARTS_WITH + '.so'
>             elif plat.startswith('darwin'):
>                 libname = Library.JLINK_SDK_STARTS_WITH + '.dylib'
>             else:
>                 self.logger.warning(f'unknown platform {plat}; assuming UNIX')
>                 libname = Library.JLINK_SDK_STARTS_WITH + '.so'
> ```

This was not obvious, at least to me: the `jink.py` module above refers to some `$ZEPHYR_BASE/scripts/west_commands/runners/jlink.py`, and not to `pylink/jlink.py`.
And changing the `JLINK_SDK_NAME` value from `libjlinkarm` to `jlinkarm` does indeed break the `JLinkBinaryRunner.jlink_version()` function implementation *there*.

I agree that substituting `JLINK_SDK_STARTS_WITH` to `JLINK_SDK_NAME` in the west command implementation should fix the regression, but:
- the corresponding patch applies to the Zephyr project, no to the pylink source code
- to begin with, this `jlink_version()` function should not (try to) replicate a subset of the logic implemented by the pylink library, but just rely on its explicitly **public** API (as far as we can enforce this kind of *contract* with Python), which I don't think `JLINK_SDK_NAME`, `JLINK_SDK_STARTS_WITH`, nor `get_appropriate_windows_sdk_name()` actually belong to

Another grep seems to confirm there aren't other references to the (private) *constant* `Library.JLINK_SDK_NAME`:
``` sh
grep -R -F "JLINK_SDK_NAME" $ZEPHYR_BASE
$ZEPHYR_BASE/scripts/west_commands/runners/jlink.py:  libname = Library.JLINK_SDK_NAME + '.so'
$ZEPHYR_BASE/scripts/west_commands/runners/jlink.py:  libname = Library.JLINK_SDK_NAME + '.dylib'
$ZEPHYR_BASE/scripts/west_commands/runners/jlink.py:  libname = Library.JLINK_SDK_NAME + '.so'
``` 

@asarbs , IIUC *your zephyr-os stop working* when you run something like `west {flash,debug} --runner jlink`, which would complain about some hard to find `jlinkarm.so` file (instead of `libjlinkarm.so`).
As a temporary workaround, and depending on your setup/requirements, you could try the default Zephyr-SDK runner for JLink devices (`nrfjprog`), or the `pyocd` runner: they both work fine while also loading the JLink DLL with the pylink library.

@hkpeprah, I really think this is a downstream (Zephyr) issue, that should be resolved *there*: 
- may be first apply a patch like the one @asarbs has suggested, to quickly fix the regression for Zephyr-SDK users relying on the `jlink` runner
- eventually rewrite the `jlink_version()` function (AFAICT, `pylink.Library()` might do the job just fine, without twisting, but I may miss something)

If you have not already planed to, and @asarbs hasn't yet done, I can open an issue [*there*](https://github.com/zephyrproject-rtos/zephyr/issues) to discuss the better approach with the Zephyr peoples, and propose our help.

Thanks.

--
chris